module/platform/darwin/SensorFilter/src/SensorFilter.cpp

```diff
@@ -87,6 +88,44 @@ namespace module::platform::darwin {
         return s.str();
     }
 
+    /// @brief Calculates the mean state, by averaging the vectors and calculating the mean quaternion
+    /// @param states The states being averaged
+    MotionModel<double>::StateVec calculateMeanState(std::vector<MotionModel<double>::StateVec> states) {
+
+        // We'll add the vectors to a zero-initialised state, then coefficient-wise divide them by the number of states
+        MotionModel<double>::StateVec mean_state{};
+        // We'll make a vector of the quaternions to take the mean of them
+        std::vector<Eigen::Quaterniond> rotations{};
+
+        for (const auto& state : states) {
+            // clang-format off
+            mean_state.rTWw          += state.rTWw;
+            mean_state.vTw           += state.vTw;
+            mean_state.omegaTTt      += state.omegaTTt;
+            mean_state.omegaTTt_bias += state.omegaTTt_bias;
+            // clang-format on
+            rotations.push_back(state.Rwt);
+        }
+
+        // clang-format off
+        mean_state.rTWw     /= states.size();
+        mean_state.vTw      /= states.size();
+        mean_state.omegaTTt /= states.size();
+        mean_state.omegaTTt /= states.size();
+        // clang-format on
+        mean_state.Rwt = utility::math::quaternion::mean(rotations.begin(), rotations.end());
+
+        return mean_state;
+    }
+
+    MotionModel<double>::StateVec sensorsToState(DarwinSensors sensors) {
+        MotionModel<double>::StateVec state;
+
+        // TODO: I have no idea how to construct a state from sensors
+        // maybe look at sims/kalman_filter for ideas
+        return state;
+    }
```

Comment:
I think the best way to do this is to actually take measurements (calling `motionFilter.measure()`). 

Might be better to average the sensor readings to get a better idea on initial system state and use these readings to initialise the filter rather than using the mean values from the config file. Might also be able to use these averages to get an idea on noise values?